src/components/pages/address/address-transactions.tsx

```diff
@@ -1,13 +1,66 @@
 import { Address } from "viem";
+import { prisma } from "@/lib/prisma";
 import { Card, CardHeader, CardContent } from "@/components/ui/card";
+import { fromPrismaTransactionWithReceipt, TransactionWithReceipt } from "@/lib/types";
+import { loadFunctions } from "@/lib/signatures";
+import TxsTable from "@/components/lib/txs-table";
 
-const AddressTransactions = ({ address }: { address: Address }) => {
+const TRANSACTION_LIMIT = 25;
+interface TransactionQueryResult {
+  transactions: TransactionWithReceipt[];
+  totalCount: number;
+}
+
+const fetchTransactionsData = async (address: Address): Promise<TransactionQueryResult> => {
+  const associatedTransactions = await prisma.transaction.findMany({
+    where: {
+      OR: [
+        { from: { contains: address } },
+        { to: { contains: address } },
+      ],
+    },
```

Comment:
You're missing a `take: Number(process.env.NEXT_PUBLIC_TXS_PER_PAGE)` here to only fetch the correct amount of transactions.